src/main/modules/oidc/index.ts

```diff
@@ -76,9 +77,21 @@ export class OidcMiddleware {
           res.locals.isLoggedIn = true;
           req.locals.api = getCaseApi(req.session.user, req.locals.logger);
           if (!req.session.userCase) {
-            req.session.userCase = await req.locals.api.getOrCreateCase(res.locals.serviceType, req.session.user);
+            const userCase = await req.locals.api.getCase();
+            if (userCase) {
+              req.session.userCase = userCase;
+            }
+          }
+          if (req.path !== HOME_URL && !req.path.startsWith(APPLYING_WITH_URL)) {
+            // Commented out restricting case creation to happen only from Applying-With page submission
+            /* if (!req.session.userCase) {
+              req.session.userCase = await req.locals.api.createCase(res.locals.serviceType, req.session.user);
+            } */
+
+            return next();
+          } else {
+            return next();
           }
```

Comment:
We don't need this if-else block, as in both we are doing same action

Reply:
Hi @cshiwarale Updated the code as per review comments.